runtime/runtime/src/congestion_control.rs

```diff
@@ -491,10 +490,11 @@ impl DelayedReceiptQueueWrapper {
             true => {
                 let metadata =
                     StateStoredReceiptMetadata { congestion_gas: gas, congestion_size: size };
-                let receipt = StateStoredReceipt::new_borrowed(receipt, metadata);
+                // FIXME eagr
+                let receipt = StateStoredReceipt::new(receipt.clone(), metadata);
```

Comment:
Are these clones gonna be deal breaker? @wacban

Reply:
It would be great if we could avoid cloning receipts. Cloning a receipt is a dangerous operation because it duplicates the operation that will be performed when the receipt is executed. For example duplicating a transfer receipt could lead to double spending.
Ideally the receipts would never be cloned. Pushing a receipt onto a queue should consume the receipt. This way the compiler would ensure that a receipt is never cloned and receipts are never duplicated.

I'm not a fan of cloning as a solution for Cow, is it possible to refactor the code so that the receipts are consumed?

Reply:
> I'm not a fan of cloning as a solution for Cow, is it possible to refactor the code so that the receipts are consumed?

I was entirely not sure those either, kinda wanna go the naive way first to see what it breaks to gain a better idea of why Cow was used in the first place. Yeah, I'll see what I could do

Reply:
If the receipts can't be consumed because information about them is needed later, maybe it would be possible to save the information and consume the receipt? Like having a queue which consumes the `Receipt` and returns `ReceiptInfo` that is used later. `ReceiptInfo` would contain the needed information, while not being a cloned receipt that can be executed.

I don't remember why exactly there was a problem with consuming them.

Reply:
Agree with jancio, the clones were introduced precisely in order to avoid clone on receipts. 

Agree with jancio again, the receipt should be consumed, I think only some metadata about it is needed after it's been stored in the StateStoredReceipt. That would require some refactoring. 

@eagr It's a great first step! I think now you can just make the receipt consumed by removing reference, see what breaks and refactor as needed. IIRC there was only one such place, where we need some simple stats about local receipts.